gdx/src/com/badlogic/gdx/utils/Scaling.java

```diff
@@ -34,9 +32,7 @@ public Vector2 apply (float sourceWidth, float sourceHeight, float targetWidth,
 			float targetRatio = targetHeight / targetWidth;
 			float sourceRatio = sourceHeight / sourceWidth;
 			float scale = targetRatio > sourceRatio ? targetWidth / sourceWidth : targetHeight / sourceHeight;
-			temp.x = sourceWidth * scale;
-			temp.y = sourceHeight * scale;
-			return temp;
+			return new Vector2(sourceWidth * scale, sourceHeight * scale);
```

Comment:
This creates a new `Vector2` on every invocation, hmmm.

Reply:
let's look into it @dar-dev

Reply:
my bad, the _apply()_ method is called multiple times. I have to use thread-local here